pandas/core/internals/managers.py

```diff
@@ -1865,11 +1902,13 @@ def _shape_compat(x):
     return stacked, placement
 
 
-def _interleaved_dtype(blocks):
+def _interleaved_dtype(blocks, allow_extension=False):
     if not len(blocks):
```

Comment:
add docstring, should this be handled at a higher level? (e.g. what case is this handling)?

Reply:
Added. This supports `df.iloc[0]` returning something other than `object` when the DataFrame is all extension arrays of the same type. I think this is worth supporting, and not *too* much work.

I've removed the `allow_extension` flag. Everywhere we call interleaved_dtype seemingly is able to handle ExtensionDtype correctly.